osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/AbstractGetPowerQualityProfileHandler.java

```diff
@@ -293,16 +330,11 @@ private boolean isDefinitionOfSameObject(final CosemObjectDefinitionDto cosemObj
         final byte[] obisBytesCosemObjectDefinition = cosemObjectDefinitionDto.getLogicalName().toByteArray();
         final byte attributeIndexCosemObjectDefinition = (byte) cosemObjectDefinitionDto.getAttributeIndex();
         final int dataIndexCosemObjectDefinition = cosemObjectDefinitionDto.getDataIndex();
-
         final int classIdCaptureObjectDefinition = captureObjectDefinition.getClassId();
         final byte[] obisBytesCaptureObjectDefinition = captureObjectDefinition.getLogicalName().toByteArray();
         final byte attributeIndexCaptureObjectDefinition = captureObjectDefinition.getAttributeIndex();
-        final int dataIndexCaptureObjectDefinition;
-        if (captureObjectDefinition.getDataIndex() == null) {
-            dataIndexCaptureObjectDefinition = 0;
-        } else {
-            dataIndexCaptureObjectDefinition = captureObjectDefinition.getDataIndex();
-        }
+        final int dataIndexCaptureObjectDefinition =
+                captureObjectDefinition.getDataIndex() == null ? 0 : captureObjectDefinition.getDataIndex();
 
         return classIdCaptureObjectDefinition == classIdCosemObjectDefinition && Arrays
```

Comment:
I would expect this method to be redundant, if equals() was defined properly on these DTO's and/or one can be mapped to the other.